redisinsight/api/src/modules/database-recommendation/providers/database-recommendation.provider.ts

```diff
@@ -0,0 +1,79 @@
+import { Injectable, Logger } from '@nestjs/common';
+import { DatabaseRecommendationEntity }
+  from 'src/modules/database-recommendation/entities/database-recommendation.entity';
+import { InjectRepository } from '@nestjs/typeorm';
+import { Repository } from 'typeorm';
+import { plainToClass } from 'class-transformer';
+import { RecommendationDto, RecommendationsDto } from 'src/modules/database-recommendation/dto';
+import { ClientMetadata } from 'src/common/models';
+
+@Injectable()
+export class DatabaseRecommendationProvider {
+  private readonly logger = new Logger('DatabaseRecommendationProvider');
+
+  constructor(
+    @InjectRepository(DatabaseRecommendationEntity)
+    private readonly repository: Repository<DatabaseRecommendationEntity>,
+  ) {}
+
+  /**
+   * Save entire entity
+   * @param databaseId
+   * @param recommendationName
+   */
+  async create(databaseId: string, recommendationName: string): Promise<RecommendationDto> {
```

Comment:
what we return here? - Promise<RecommendationDto>. (should be Recommendation)

Reply:
I'm not sure that caught you. Here we return Promise<DatabaseRecommendation>

Reply:
this.repository.save - returns **Promise\<DatabaseRecommendationEntity\>** when you claim that your function should return **Promise\<RecommendationDto\>**